File: front/src/app/core/model/card-set.ts
import { CardSetDTO } from './dto/card-set-dto';

export class CardSet {
  id: number;
  name?: string;
  code?: string;
  rarity?: string;
  rarityCode?: string;
  price?: number;
  cardId?: number;
  cssRarityCode: string;

  constructor(cardSetDTO: CardSetDTO) {
    this.id = cardSetDTO.id;
    this.name = cardSetDTO.name;
    this.code = cardSetDTO.code;
    this.rarity = cardSetDTO.rarity;
    this.rarityCode = cardSetDTO.rarityCode;
    this.price = cardSetDTO.price;
    this.cardId = cardSetDTO.cardId;
    this.cssRarityCode = cardSetDTO.rarityCode?.replace('(', '').replace(')', '') || '';
  }
}

File: back/src/main/java/com/skytrix/model/entity/CardSet.java
package com.skytrix.model.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class CardSet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String code;
    private String rarity;
    private String rarityCode;
    private float price;
    @ManyToOne
    private Card card;

}
